analyzer-comments/java/annalyns-infiltration/avoid_unnecessary_parenthesis.md

```diff
@@ -0,0 +1,11 @@
+# Avoid unnecessary parenthesis
+
+If you use an `||` expression to OR two smaller expressions and either expression is surrounded by parentheses and only ANDs some terms together (e.g. `knightIsAwake || (archerIsAwake && !prisonerIsAwake)`), the parentheses is unnecessary because `&&` has the higher precedence over `||`:
```

Comment:
```suggestion
If you use an `||` expression to OR two smaller expressions and either expression is surrounded by parentheses and only ANDs some terms together (e.g. `knightIsAwake || (archerIsAwake && !prisonerIsAwake)`), the parentheses can be omitted because `&&` has the higher precedence over `||`:
```

And then I would do this.